### ISSUE TITLE
Multiple Inner Loops

### ISSUE THREAD

**Issue by pharringtonp19** [facebookresearch/functorch#25]
Great project -- 

What would be the most straightforward way to allow for multiple inner training loops?

Specifically with regards to the MAML example, how could I allow for a user defined number of inner loops?

```
def get_loss_for_task(x1, y1, x2, y2):
        def inner_loss(params, x1, y1):
            f = net(params, (x1,))
            loss = mse_loss(f, y1)
            return loss

        grads = grad(inner_loss)(params, x1, y1)
        new_params = [(params[i] - alpha*grads[i]) for i in range(len(params))]

        v_f = net(new_params, (x2,))
        return mse_loss(v_f, y2)
```

As a note, after changing  `losses.append(loss2)` to `losses.append(loss2.item())` I was able to plot the results

**Comment by Chillee** [comment]
`get_loss_for_task(x1, y1, x2, y2)` gets you the loss for a single inner task (i.e. a single "inner loop", although it's only doing one step of the inner training loop here). This single inner task is `(x1, y1, x2, y2)`, and consists of 2 examples pairs. So, let's say that `x1` has shape`[5]`.

So, if you vmap over `get_loss_for_task`, then you'd now pass in `[B, 5]` for x1 instead of just `[5]`. This corresponds to doing 5 runs of `get_loss_for_task`, or running 5 "inner training loops" in parallel.

**Comment by pharringtonp19** [comment]
Thanks for the response, I should have been clearer in my original question.

As you note, "it's only doing one step of the inner training loop here". How would I adjust the code so that it does `k` steps  instead of 1?

**Comment by Chillee** [comment]
https://github.com/zou3519/functorch/blob/main/examples/maml_regression/evjang_transforms.py#L73

Instead of doing only one gradient step here, you need to pass in multiple examples and do multiple gradient steps in a loop.

**Comment by pharringtonp19** [comment]
@Chillee  Thanks
